pkg/istio/authorizationpolicy/controller_test.go

```diff
@@ -123,30 +154,300 @@ func newFakeController(services []*v1.Service, fake bool, stopCh <-chan struct{}
+				assert.Nil(t, err, "add athenz domain crd to cache should not return error")
+				key, err := cache.MetaNamespaceKeyFunc(tt.item.Resource)
+				assert.Nil(t, err, "function convert item interface to key should not return error")
+				err = c.sync(key)
+				assert.Nil(t, err, "sync function should not return error")
+
+				// Updated Spec should be equal
+				genAuthzPolicy := c.configStoreCache.Get(collections.IstioSecurityV1Beta1Authorizationpolicies.Resource().GroupVersionKind(), tt.expectedAuthzPolicy.Name, tt.expectedAuthzPolicy.Namespace)
+				// set creation timestamp and resource version for expected authz policy resource, which are generated on the fly
+				tt.expectedAuthzPolicy.ConfigMeta.CreationTimestamp = genAuthzPolicy.ConfigMeta.CreationTimestamp
+				tt.expectedAuthzPolicy.ConfigMeta.ResourceVersion = genAuthzPolicy.ConfigMeta.ResourceVersion
+				assert.Equal(t, tt.expectedAuthzPolicy, *genAuthzPolicy, "created authorization policy spec should be equal")
+			}
+		})
+	}
+}
+
+func TestSyncAuthzPolicy(t *testing.T) {
+	tests := []struct {
+		name                string
+		inputService        *v1.Service
+		inputAthenzDomain   *adv1.AthenzDomain
+		initAuthzPolicySpec model.Config
+		item                Item
+		expectedAuthzPolicy model.Config
+	}{
+		{
+			name:                "when there is manual modification of authz policy resource, controller will revert back to spec matched with athenz domain crd",
+			inputService:        onboardedService,
+			inputAthenzDomain:   onboardedAthenzDomain,
+			initAuthzPolicySpec: getOldAuthzPolicy(),
+			item:                Item{Operation: model.EventUpdate, Resource: getModifiedAuthzPolicy()},
+			expectedAuthzPolicy: getExpectedAuthzPolicy(),
+		},
+		{
+			name:                "when there is deletion of authz policy resource, controller will recreate the authz policy",
+			inputService:        onboardedService,
+			inputAthenzDomain:   onboardedAthenzDomain,
+			initAuthzPolicySpec: getOldAuthzPolicy(),
+			item:                Item{Operation: model.EventDelete, Resource: getModifiedAuthzPolicy()},
+			expectedAuthzPolicy: getExpectedAuthzPolicy(),
+		},
+		{
+			name:                "when there is manual modification of authz policy resource with override annotation, controller should do nothing",
+			inputService:        onboardedService,
+			inputAthenzDomain:   onboardedAthenzDomain,
+			initAuthzPolicySpec: getModifiedAuthzPolicyCRWithOverrideAnnotation(),
+			item:                Item{Operation: model.EventUpdate, Resource: getModifiedAuthzPolicyWithOverrideAnnotation()},
+			expectedAuthzPolicy: getModifiedAuthzPolicyCRWithOverrideAnnotation(),
+		},
+		{
+			name:                "when there is manual creation of authz policy without override annotation, controller should delete this create resource",
+			inputService:        notOnboardedServiceWithAnnotation,
+			inputAthenzDomain:   onboardedAthenzDomain,
+			initAuthzPolicySpec: getOldAuthzPolicy(),
```

Comment:
shouldn't this be set to `model.Config{}` since the authz policy is being manually created